### PR TITLE
ci: Stop using spot instances

### PR DESCRIPTION
**Pull request by gerhard**
> [!NOTE]
> Already includes changes from https://github.com/dagger/dagger/pull/8643

They are getting disrupted often, and that becomes annoying after a few weeks of turbulence.

Before we can merge this, need to introduce `dind-od` instances.

### CODE REVIEW

**Review by jedevc — APPROVED** [review]
LGTM once we have the new runners setup :tada: Approving!

**Comment by gerhard** [comment]
I still need to introduce new `dind` instance via a PR to the private repository.

Wait for @shykes to approve before merging. Related to:
- https://github.com/dagger/dagger/pull/8355

**Review by matipan — APPROVED** [review]

**Comment by matipan** [comment]
With https://github.com/dagger/dagger.io/pull/4003 all runners are now on-demand. Good to merge!

**Pull request merged**
